include/api/CSingleStreamSearcher.h

```diff
@@ -38,13 +38,19 @@ class API_EXPORT CSingleStreamSearcher : public core::CDataSearcher {
     //! called.
     CSingleStreamSearcher(const TIStreamP& stream);
 
+    //! Ensure destructor is called to consume stream to its end
+    ~CSingleStreamSearcher();
+
     //! Get the stream to retrieve data from.
     //! \return Pointer to the input stream.
     //! Some errors cannot be detected by this call itself, and are
     //! indicated by the stream going into the "bad" state as it is
     //! read from.
     virtual TIStreamP search(size_t currentDocNum, size_t limit);
```

Comment:
```suggestion
    TIStreamP search(size_t currentDocNum, size_t limit) override;
```

(because the compiler will moan if `override` is used for one overridden virtual function and `virtual` for a different one in the same class)